pkg/api/types.go

```diff
@@ -472,6 +479,22 @@ type FCVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// FlexVolume represents a generic volume resource that is
+// provisioned/attached using a exec based plugin.
+type FlexVolumeSource struct {
+	// Driver is the name of the driver to use for this volume.
+	Driver string `json:"driver"`
+	// Required: Filesystem type to mount.
+	// Must be a filesystem type supported by the host operating system.
+	// Ex. "ext4", "xfs", "ntfs"
+	FSType string `json:"fsType,omitempty"`
+	// Optional: Defaults to false (read/write). ReadOnly here will force
+	// the ReadOnly setting in VolumeMounts.
+	ReadOnly bool `json:"readOnly,omitempty"`
+	// Optional: Extra driver options if any.
+	Options map[string]string `json:"options,omitempty"`
```

Comment:
you may also need a `secret_option Secret` where you want to store your e.g. `cifs` credentials used in your example.